### PR TITLE
FABJ-536 Add support for mutual TLS when connecting to CA

### PR DESCRIPTION
**Pull request by TsvetanG**

### CODE REVIEW

File: src/main/java/org/hyperledger/fabric_ca/sdk/HFCAClient.java
Comment (bestbeforetoday) [inline src/main/java/org/hyperledger/fabric_ca/sdk/HFCAClient.java:1634]: I'm being very picky but it's worth getting it right... `tlsCleintCertificate` should be `tlsClientCertificate`

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]
This needs to have accompanying tests; ideally both unit tests (where practical) and certainly used in integration tests

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]

**Comment by bestbeforetoday** [comment]
I know I'm raising lots of comments but thank you for taking the time to submit these changes. We appreciate it!

**Comment by TsvetanG** [comment]
@bestbeforetoday Thank you for the feedback and comments! I have addressed all of them. Any further comments are welcome!
As my team is using HLF 2.2 we will need this change to be incorporated in 2.2.x branch. Let me know if I can help to make that happen.

Best Regards,
Tsvetan
PS: Apology for the delay!

**Review by bestbeforetoday — APPROVED** [review]

**Comment by bestbeforetoday** [comment]
The changes look good to me but they won't merge as-is since the base branch has changed. It needs to be rebased on top of the current branch state.

Feel free to cherry-pick this change to both `release-2.2` and `master` branches and submit PRs for those changes too. Otherwise I should be able to do the cherry-pick on your behalf. If your intended target is only the `release-2.2` branch then you could submit a PR for only `master` and `release-2.2` branches (preferably `master` first) and leave `release-1.4` as-is.

**Comment by yzhivkov** [comment]
Pushed one more commit so that the code is streamlined with master. For some reason one of the tests is failing but does not seem to be caused by the changes in this PR. Is there any way we can trigger a rebuild?

**Review by bestbeforetoday — APPROVED** [review]

**Comment by bestbeforetoday** [comment]
This PR is out of date with the current release-1.4 branch so needs to be rebased on release-1.4 before merging.

**Pull request merged**
